### DISCUSSION TITLE
Load option of BelongTo field based on BelongTo select value

### DISCUSSION THREAD

**Discussion by benjy8001** [moonshine-software/moonshine#1339] [Q&A] [answered]
Hi ,

Is there a way to dynamicaly load options of a BelongTo field based on the selected option value of an other BelongTo field?

I try with reactive, but with success.

**Comment by lee-to** [comment]
use asyncSearch

**Reply by benjy8001** [reply]
Hi @lee-to , ok, thanks.
Is there a way to trigger an asyncSearch after the selection in the first BelongsTo Field, in order to not force the user to go to the field and search on it manualy ?

**Reply by lee-to** [reply]
look at the new force mode and the fact that the asyncSearch callback also receives the entire current request with the form data

**Reply by benjy8001** [reply]
Hi @lee-to , unfortunatly, i didn't see the force mode option . Can you tell me were is it ?

Is there a way to trigger an event in order to auto refresh the second BelongsTo Field ?

Thanks in advance !

**Comment by benjy8001** [comment, marked as answer]
Resolve by using `->associatedWith`

Exemple : 
```
            BelongsTo::make(__('moonshine::ui.resource.makes.label'), relationName: 'carMake')
                ->required()
                ->searchable()
                ->sortable(),
            BelongsTo::make(__('moonshine::ui.resource.models.label'), relationName: 'carModel')
                ->associatedWith('car_make_id')
                ->required()
                ->searchable()
                ->sortable(),
```
